src/components/organisms/ProfileModal/UserInformationContent/UserInformationContent.tsx

```diff
@@ -1,179 +1,214 @@
+  currentVenueSelectorData,
+} from "utils/selectors";
 import { venueLandingUrl } from "utils/url";
-import { currentVenueSelector } from "utils/selectors";
 
-import { useSelector } from "hooks/useSelector";
-import { useUser } from "hooks/useUser";
 import { useVenueId } from "hooks/useVenueId";
+import { useSelector } from "hooks/useSelector";
 
 import { updateUserProfile } from "pages/Account/helpers";
+
 import { Badges } from "components/organisms/Badges";
 import { UserStatusDropdown } from "components/atoms/UserStatusDropdown";
+import { Button } from "components/atoms/Button";
 import { UserAvatar } from "components/atoms/UserAvatar";
 
-interface PropsType {
-  setIsEditMode: (value: boolean) => void;
-  setIsPasswordEditMode: (value: boolean) => void;
-  hideModal: () => void;
+import editIcon from "assets/icons/profile-edit-icon.svg";
+
+import { UserProfileMode } from "../ProfilePopoverContent";
+
+import "./UserInformationContent.scss";
+
+export interface UserInformationContentProps {
+  setUserProfileMode: (value: UserProfileMode) => void;
+  setProfileLinkToEdit: (profileLink?: ProfileLink) => void;
+  user?: WithId<User>;
+  email?: string | null;
 }
 
-const UserInformationContent: React.FunctionComponent<PropsType> = ({
-  setIsEditMode,
-  setIsPasswordEditMode,
-  hideModal,
+export const UserInformationContent: React.FunctionComponent<UserInformationContentProps> = ({
+  setUserProfileMode,
+  setProfileLinkToEdit,
+  user,
+  email,
 }) => {
-  const { user, profile, userWithId } = useUser();
+  const profileQuestions = useSelector(
+    (state) => currentVenueSelectorData(state)?.profile_questions
+  );
   const venueId = useVenueId();
   const venue = useSelector(currentVenueSelector);
 
-  const profileQuestions = venue?.profile_questions;
-
   const history = useHistory();
   const firebase = useFirebase();
 
   const logout = useCallback(async () => {
     await firebase.auth().signOut();
 
-    // we need to hide the modal because if we already are on the Entrance Page, history.push has no effect
-    hideModal();
     history.push(IS_BURN ? "/enter" : venueId ? venueLandingUrl(venueId) : "/");
-  }, [firebase, hideModal, history, venueId]);
+  }, [firebase, history, venueId]);
 
   const toggleKidsMode = useCallback(async () => {
-    if (user && profile) {
-      profile.kidsMode = !profile?.kidsMode;
-      await updateUserProfile(user.uid, { kidsMode: profile.kidsMode });
+    if (user) {
+      user.kidsMode = !user?.kidsMode;
+      await updateUserProfile(user.id, { kidsMode: user.kidsMode });
     }
-  }, [profile, user]);
+  }, [user]);
 
   const toggleAnonMode = useCallback(async () => {
-    if (user && profile) {
-      profile.anonMode = !profile?.anonMode;
-      await updateUserProfile(user.uid, { anonMode: profile.anonMode });
+    if (user) {
+      user.anonMode = !user?.anonMode;
+      await updateUserProfile(user.id, { anonMode: user.anonMode });
     }
-  }, [profile, user]);
+  }, [user]);
```

Comment:
What is the intent of this change?

Reply:
⚠️ This is mutating `user.anonMode`, which is not ok. This is a blocker. This cannot be merged in its current state.